src/main/java/com/smartagilify/core/aspect/BaseAspect.java

```diff
@@ -0,0 +1,69 @@
+package com.smartagilify.core.aspect;
+
+import com.smartagilify.core.constant.ErrorConstant;
+import com.smartagilify.core.constant.RestAddress;
+import com.smartagilify.core.model.BaseDTO;
+import com.smartagilify.core.model.ResultDTO;
+import com.smartagilify.core.util.HashUtil;
+import jakarta.servlet.http.HttpServletRequest;
+import lombok.extern.slf4j.Slf4j;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Before;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+import java.security.InvalidParameterException;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Slf4j
+@Aspect
+@Component
+public class BaseAspect {
+
+    protected final Pattern controllerRestPattern = Pattern.compile(RestAddress.REST_PREFIX + "/(.*?)/", Pattern.DOTALL);
+
+    @Before(value = "execution(* com.smartagilify.core.controllers.*.*ById*(Long, String)) && args(id, hashcode)", argNames = "id,hashcode")
+    public void checkIdAndHashCode(Long id, String hashcode) {
+        ServletRequestAttributes attributes = (ServletRequestAttributes) RequestContextHolder.getRequestAttributes();
+        if (attributes == null) throw new InvalidParameterException(ErrorConstant.INVALID_REQUEST_PARAMETER);
+        HttpServletRequest request = attributes.getRequest();
+
+        String expectedHash = HashUtil.createIdHashcode(id, getRestControllerName(request), -1);
```

Comment:
as you have mentioned in the TODO clause below there is no user id present to include in the hash operation

Reply:
yes but finally we gotta fill this field